lib/ansible/modules/cloud/ovirt/ovirt_vms.py

```diff
@@ -541,13 +552,61 @@ def __get_template_with_version(self):
 
         return template
 
+    def __get_storage_domain_and_all_template_disks(self, template):
+
+        storage_domain = self._module.params.get('storage_domain')
+
+        storage_domain_obj = search_by_name(
+            self._connection.system_service().storage_domains_service(),
+            storage_domain,
+        )
+
+        disk_attachments = (
+            self._connection.follow_link(template.disk_attachments)
+        )
+        disks = list()
+
+        disk_format = self._module.params.get('disk_format')
+        if disk_format == "RAW":
+            format = otypes.DiskFormat.RAW
+        else:
+            format = otypes.DiskFormat.COW
+
+        for disk in disk_attachments:
+            disks.append(
+                otypes.DiskAttachment(
+                    disk=otypes.Disk(
+                        id=disk.id,
+                        format=otypes.DiskFormat.COW,
+                        storage_domains=[
+                            otypes.StorageDomain(
+                                id=storage_domain_obj.id
+                            )
+                        ]
+                    )
+                )
+            )
+
+        return storage_domain_obj, disks
+
+
     def build_entity(self):
         template = self.__get_template_with_version()
+
+        disks_attachments = None
```

Comment:
Can you move this logic to `__get_storage_domain_and_all_template_disks` and just call:
```
return otypes.Vm(
...
disk_attachments=self.__get_storage_domain_and_all_template_disks()
...
)
```